packages/sign-client/src/controllers/engine.ts

```diff
@@ -955,28 +954,6 @@ export class Engine extends IEngine {
       throw new Error(message);
     }
 
-    // if the length of the namespaces is greater than the length of the required namespaces
-    // then the user is trying to approve part or all of the optional namespaces so we need to validate
-    if (Object.keys(namespaces).length > Object.keys(proposal.requiredNamespaces).length) {
-      // filter out the optional namespaces that are not being used
-      const namespacesToValidate = Object.keys(proposal.optionalNamespaces).filter(
-        (namespace) => namespaces[namespace],
-      );
-      const usedOptionalNamespaces = {};
-      for (const key in proposal.optionalNamespaces) {
-        if (namespacesToValidate.includes(key)) {
-          usedOptionalNamespaces[key] = proposal.optionalNamespaces[key];
-        }
-      }
-      const conformingNamespacesError = isConformingNamespaces(
```

Comment:
A lot going on here so not immediately clear to me why we can kick this whole block of validation out, I'm guessing it's too rigid based on the now more dynamic parsing approach?

Reply:
I removed it because we don't really need to check the `optional` namespaces. As long as the structure is correct (types responsibility) we're OK

Reply:
👍🙇‍♂️